test/methods/gd_lipschitz_approximation.jl

```diff
@@ -0,0 +1,172 @@
+# OptimizationMethods.jl
+
+module TestGDLipschitzApproximation
+
+using Test, OptimizationMethods, LinearAlgebra, Random
+
+@testset "Method: GD Lipschitz Approximation" begin
+
+    #Testing Context 
+    Random.seed!(1010)
+
+    ##########################################
+    # Test struct properties
+    ##########################################
+
+    ## test if method struct is defined 
+    @test @isdefined LipschitzApproxGD
+    
+    ## test supertype of method struct 
+    @test supertype(LipschitzApproxGD) == 
+        OptimizationMethods.AbstractOptimizerData
+
+    ## Test Field Names and Types 
+    field_info(type::T) where T = [
+        [:name, String], 
+        [:init_stepsize, type], 
+        [:threshold, type], 
+        [:max_iterations, Int64],
+        [:iter_diff, Vector{type}],
+        [:grad_diff, Vector{type}],
+        [:iter_hist, Vector{Vector{type}}],
+        [:grad_val_hist, Vector{type}],
+        [:stop_iteration, Int64]
+    ]
+
+    float_types = [Float16, Float32, Float64]
+    param_dim = 10
+
+    for float_type in float_types
+        fields = field_info(float_type)
+        optData = LipschitzApproxGD(
+            float_type,
+            x0 = zeros(float_type, param_dim), 
+            init_stepsize = float_type(1e-4), 
+            threshold = float_type(1e-4),
+            max_iterations = 100
+        )
+
+        for field_elem in fields
+            #Test Names (does not depend on field_type)
+            @test field_elem[1] in fieldnames(LipschitzApproxGD)
+
+            #Test Types
+            @test field_elem[2] == typeof(getfield(optData, field_elem[1]))
+        end
+    end
+
+    ## Test Assertions
+    @test_throws AssertionError LipschitzApproxGD(
```

Comment:
I think this should be in the for loop above

Reply:
Done